src/Repository/RepositoryTrait.php

```diff
@@ -15,4 +17,15 @@ public function saveChanges(): void
         $this->getEntityManager()->flush();
     }
 
+    public function delete(object $entity): bool
+    {
+        try {
+            $entityManager = $this->getEntityManager();
+            $entityManager->remove($entity);
+            $entityManager->flush();
+            return true;
+        } catch (ORMException $e) {
+            throw new ORMException("Error: Entity could not be deleted.");
```

Comment:
why throw an ORMException? If you already catch it, throw your custom Exeception